course_catalog/etl/loaders.py

```diff
@@ -212,5 +224,160 @@ def load_video(video_data):
+    with transaction.atomic():
+        for position, video in enumerate(videos):
+            PlaylistVideo.objects.update_or_create(
+                playlist=playlist, video=video, defaults={"position": position}
+            )
+        PlaylistVideo.objects.filter(playlist=playlist).exclude(
+            video_id__in=[video.id for video in videos]
+        ).delete()
+
+    return playlist
+
+
+def load_playlists(video_channel, playlists_data):
+    """
+    Load a list of channel playlists
+
+    Args:
+        video_channel (VideoChannel): the video channel instance this playlist is under
+        playlists_data (iter of dict): iterable of the video playlists
+
+
+    Returns:
+        list of Playlist:
+            the created or updated playlists
+    """
+    playlists = [
+        load_playlist(video_channel, playlist_data) for playlist_data in playlists_data
+    ]
+    playlist_ids = [playlist.id for playlist in playlists]
+
+    # remove playlists that no longer exist
+    Playlist.objects.filter(channel=video_channel).exclude(id__in=playlist_ids).update(
+        published=False
+    )
+
+    return playlists
+
+
+def load_video_channel(video_channel_data):
+    """
+    Load a single video channel
+
+    Arg:
+        video_channel_data (dict):
+            the normalized video channel data
+    Returns:
+        VideoChannel:
+            the updated or created video channel
+    """
+    platform = video_channel_data.pop("platform")
+    channel_id = video_channel_data.pop("channel_id")
+    playlists_data = video_channel_data.pop("playlists", [])
+    topics_data = video_channel_data.pop("topics", [])
+    offered_by_data = video_channel_data.pop("offered_by", [])
+
+    video_channel, _ = VideoChannel.objects.update_or_create(
+        platform=platform, channel_id=channel_id, defaults=video_channel_data
+    )
+
+    load_topics(video_channel, topics_data)
+    load_offered_bys(video_channel, offered_by_data)
+    load_playlists(video_channel, playlists_data)
+
+    return video_channel
+
+
+def load_video_channels(video_channels_data):
+    """
+    Load a list of video channels
+
+    Args:
+        video_channels_data (iter of dict): iterable of the video channels data
+
+    Returns:
+        list of VideoChannel:
+            list of the loaded videos
+    """
+    video_channels = []
+
+    # video_channels_data is a generator
+    for video_channel_data in video_channels_data:
+        try:
+            video_channel = load_video_channel(video_channel_data)
+        except ExtractException:
+            # video_channel_data has lazily evaluated generators, one of them could raise an extraction error
+            # this is a small pollution of separation of concerns
+            # but this allows us to stream the extracted data w/ generators
+            # as opposed to having to load everything into memory, which will eventually fail
+            log.exception("Error with extracted video channel")
```

Comment:
Would it be useful to display part of the `video_channel_data` in the exception message to narrow down which one caused a problem?